spaceship.zsh

```diff
@@ -91,6 +91,9 @@ SPACESHIP_PROMPT_SUFFIXES_SHOW="${SPACESHIP_PROMPT_SUFFIXES_SHOW:=true}"
 SPACESHIP_PROMPT_DEFAULT_PREFIX="${SPACESHIP_PROMPT_DEFAULT_PREFIX:="via "}"
 SPACESHIP_PROMPT_DEFAULT_SUFFIX="${SPACESHIP_PROMPT_DEFAULT_SUFFIX:=" "}"
 
+# LS_COLORS
+SPACESHIP_LSCOLORS_DEFINE="${SPACESHIP_LSCOLORS_DEFINE:=true}"
```

Comment:
`SPACESHIP_LSCOLORS_DEFINE` 🤔

Option's name should be discussed. Maybe we should use just `SPACESHIP_LSCOLORS` and assign this variable default value. What do you think?

Reply:
Yeah, I'm not super fond of the name too 🙂

Maybe `SPACESHIP_USE_LSCOLORS` or `SPACESHIP_USE_DEFAULT_LSCOLORS`?

As for default value, I'm not sure that this is the best approach. Note that this theme actually defines two variables (`LS_COLORS` for Linux and `LSCOLORS` for Mac), as well as calls `zstyle ':completion:*'`, and if people use a custom provider, all of this is already done and so none of this should be executed again by the prompt theme.

Alternatively we can define `LSCOLORS` and `LS_COLORS` if and only if they are undefined yet, that way we don't overwrite anything and yet keep defining the default value for those who don't use any other provider of `LS_COLORS`. But I'm a bit concerned about this behavior, I'm not as confident that this will not break something for some people. One explicit configuration variable with `true` as default value seems like a safer bet.

-----------

To be honest, this whole `LS_COLORS` block is a bit surprising for me to see as part of the prompt scheme, usually it's done somewhere else. Providing LS_COLORS is part of either a custom rich provider (like [trapd00r/LS_COLORS](https://github.com/trapd00r/LS_COLORS)) or part of framework (like [prezto/init.zsh](https://github.com/sorin-ionescu/prezto/blob/aa0395244d3136b5c7b3495feb5c39beaff4206e/modules/utility/init.zsh#L73-L102) or [oh-my-zsh/theme-and-appearance.zsh](https://github.com/robbyrussell/oh-my-zsh/blob/a7fb55cf13bd5956a183e42916cf160a962b6370/lib/theme-and-appearance.zsh#L26-L33)), while configuring `zstyle ':completion:*'` for autocompletion coloring is part of autocompletion plugins (like [prezto/completion.zsh](https://github.com/sorin-ionescu/prezto/blob/aa0395244d3136b5c7b3495feb5c39beaff4206e/modules/completion/init.zsh#L90)) or again part of the framework (like [oh-my-zsh/theme-and-appearance.zsh](https://github.com/robbyrussell/oh-my-zsh/blob/a7fb55cf13bd5956a183e42916cf160a962b6370/lib/theme-and-appearance.zsh#L37-L38)).

Reply:
@denysdovhan since breaking changes are acceptable in `3.0`, I would really recommend to consider removing this whole `LS_COLORS` block from the spaceship theme, as I said above it doesn't feel like it belongs here.

Reply:
@maximbaz I'm working on it 😉